### PR TITLE
CURL HTTP request builder

### PR DESCRIPTION
**Pull request by sergey-shandar**

### CODE REVIEW

File: sdk/core/core/inc/az_curl_adapter.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_CURL_ADAPTER_H
+#define AZ_CURL_ADAPTER__H
```
Comment (antkmsft) [inline sdk/core/core/inc/az_curl_adapter.h]:
```suggestion
#define AZ_CURL_ADAPTER_H
```

File: sdk/core/core/inc/az_http_client.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef AZ_HTTP_CLIENT_H
+#define AZ_HTTP_CLIENT__H
```
Comment (antkmsft) [inline sdk/core/core/inc/az_http_client.h]:
```suggestion
#define AZ_HTTP_CLIENT_H
```

File: sdk/core/core/src/az_curl_adapter.c [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  // Method
+  curl_easy_setopt(p_curl->p_curl, CURLOPT_POSTFIELDS, p_request->body.begin);
+
+  // URL
+  char * url;
+  AZ_RETURN_IF_FAILED(az_http_url_to_new_str(p_request, &url));
+  curl_easy_setopt(p_curl->p_curl, CURLOPT_URL, url);
+  free(url);
+
+  curl_easy_setopt(p_curl->p_curl, CURLOPT_WRITEFUNCTION, write_to_span);
```
Comment (sergey-shandar) [inline sdk/core/core/src/az_curl_adapter.c]: `write_to_span` allocates memory but we never free this memory so it's a memory leak.

**Comment by sergey-shandar** [comment]
See PR #103

**Pull request closed without merging**